openresty/Dockerfile-alpine

```diff
@@ -0,0 +1,253 @@
+# Current latest Openresty version is here.
+# Also specify the Nginx core used. (Openresty, adds their own version numbers after the nginx one, so we can't use the same number for both openresty and nginx)
+# Also specify the LUA version used. (Comes with the latest Openresty src bundle.)
+ARG OPENRESTY_VERSION="1.21.4.2" \
+    NGINX_VERSION="1.21.4" \
+    LUA_JIT_VERSION="2.1-20230410"
+
+FROM openresty/openresty:${OPENRESTY_VERSION}-alpine-fat as build
+
+ARG MODSEC_VERSION=3.0.9 \
+    OPENRESTY_VERSION \
+    NGINX_VERSION \
+    LUA_JIT_VERSION 
+
+# Remove pcre2 support for openresty, until it gets update to pcre2.
+RUN set -eux; \
+    apk add --no-cache --virtual .build-deps \
+        autoconf \
+        automake \
+        ca-certificates \
+        coreutils \
+        curl-dev \
+        g++ \
+        gcc \
+        geoip-dev \
+        git \
+        libc-dev \
+        libfuzzy2-dev \
+        libmaxminddb-dev \
+        libstdc++ \
+        libtool \
+        libxml2-dev \
+        libxml2 \
+        libxslt-dev \
+        libxslt \
+        linux-headers \
+        lmdb-dev \
+        make \
+        patch \
+        pkgconfig \
+        pcre \
+        pcre-dev \
+        yajl-dev \
+        gd \
+        gd-dev \
+        zlib-dev
+
+
+WORKDIR /sources
+
+# Download modsecurity and compile it.
+RUN set -eux; \
+    git clone https://github.com/SpiderLabs/ModSecurity --branch v"${MODSEC_VERSION}" --depth 1 --recursive; \
+    cd ModSecurity; \
+    ARCH=$(gcc -print-multiarch); \
+    sed -ie "s/i386-linux-gnu/${ARCH}/g" build/ssdeep.m4; \
+    sed -ie "s/i386-linux-gnu/${ARCH}/g" build/pcre.m4; \
+    ./build.sh; \
+    ./configure --with-yajl --with-ssdeep --with-lmdb --with-geoip --enable-silent-rules --with-pcre=/usr/bin/pcre-config; \
+    make install; \
+    strip /usr/local/modsecurity/lib/lib*.so*
+
+# We use master Modsec version
+# Instead of Nginx download the bundled openresty for this version of openresty base image. We can use the openresty already installed on the base image and get the compile options from there. For building the nginx module we can use the bundled nginx.
```

Comment:
Why is this necessary? Doesn't the Openresty image also include nginx?

```suggestion
# Download Openresty with bundled nginx. We can then get the compilation options for nginx from the installed Openresty and then build the ModSecurity nginx connector using the bundled nginx.
```

Reply:
I have updated the comment.

I download the bundle mostly for the 'configure' script. I did not find this in the premade Openresty image under /usr/local/openresty. I guess they removed unneccessary build files from their image. As we need to add a new nginx module we need to download the version specific src of Openresty to do that.